frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.unit.spec.ts

```diff
@@ -0,0 +1,401 @@
+    });
+  });
+
+  describe("Series with preferred colors", () => {
+    const multipleSeries: SeriesWithOneOrLessDimensions[][] = [
+      [
+        {
+          name: "Sum of Total",
+          cardName: "Bar chart",
+          yAxisPosition: "left",
+          type: "bar",
+          data: [["2016-04-24T00:00:00Z", 52.75594257942132]],
+          column: {
+            name: "sum",
+            source: "aggregation",
+            display_name: "Sum of Total",
+          },
+        },
+      ],
+    ];
+
+    it("should assign colors from preferred color", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        merge(settings, { x: { type: "timeseries" } }),
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#88BF4D", // accent1 color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("should assign colors from whitelabel colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        merge(settings, {
+          x: { type: "timeseries" },
+        }),
+        getPalette({ accent1: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#123456", // whitelabel color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("should assign colors from column colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        merge(settings, {
+          x: { type: "timeseries" },
+          series_settings: {
+            sum: {
+              color: "#987654",
+            },
+          },
+        }),
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#987654", // column color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+  });
+
+  describe("Series with 2 dimension", () => {
+    const multipleSeries: SeriesWithTwoDimensions[][] = [
```

Comment:
Sampled from this chart
![image](https://user-images.githubusercontent.com/1937582/201139624-328c5849-a3fe-4f5c-9e02-0b6ee822c470.png)